Detectors/MUON/MID/Clustering/test/bench_Clusterizer.cxx

```diff
@@ -20,81 +20,112 @@
 #include "DataFormatsMID/StripPattern.h"
 #include "Clusterizer.h"
 
-std::vector<o2::mid::ColumnData> generateTestData(int deId, const o2::mid::Mapping& midMapping)
+o2::mid::ColumnData& getColumn(std::vector<o2::mid::ColumnData>& patterns, int icolumn, int deId)
+{
+  for (auto& currColumn : patterns) {
+    if (currColumn.columnId == icolumn) {
+      return currColumn;
+    }
+  }
+
+  patterns.emplace_back(o2::mid::ColumnData());
```

Comment:
Hi Diego,
hmm, though it would work with the two arguments because it's a simple struct (just checked, and it compiles in my case). Looking more into it, it seems that  member `ColumnData::patterns` is not initialized anyhow. So a constructor with 3 arguments is not needed. What should work is also
```
 patterns.emplace_back(o2::mid::ColumnData{deId, icolumn});
```
again leaving the last member uninitialized.

Ciao, Matthias

Reply:
Hi Matthias,
actually, I tried with the solution that you proposed, i.e.:
`patterns.emplace_back(deId, icolumn);`
and it did not work on my case:
`/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:1759:31: er
ror: no matching constructor for initialization of 'o2::mid::ColumnData'
            ::new((void*)__p) _Up(_VSTD::forward<_Args>(__args)...);
[....]
/Users/dstocco/Documents/projects/alice/software/sw/SOURCES/O2/o2-master/0/DataFormats/Detectors/MUON/MID/include/Da
taFormatsMID/StripPattern.h:50:8: note: candidate constructor (the implicit copy constructor) not viable: requires 1
 argument, but 2 were provided
`
Anyways, what you propose now works fine (should be equivalent to what I wrote above, except that you use an emplace_back instead of a push_back): I will adopt your solution.

Thanks,
cheers,
Diego